src/codegen/enums.rs

```diff
@@ -0,0 +1,154 @@
+use analysis::namespaces;
+use case::CaseExt;
+use codegen::general::{self, version_condition, version_condition_string};
+use env::Env;
+use file_saver;
+use library::*;
+use std::collections::HashSet;
+use std::io::prelude::*;
+use std::io::Result;
+use std::path::Path;
+use traits::*;
+use version::Version;
+
+pub fn generate(env: &Env, root_path: &Path, mod_rs: &mut Vec<String>) {
+    let path = root_path.join("enums.rs");
+    mod_rs.push("\nmod enums;".into());
+    file_saver::save_to_file(path, env.config.make_backup, |w| {
+        try!(general::start_comments(w, &env.config));
+        try!(writeln!(w, "{}", r#"
+use ffi;
+use glib_ffi;
+use glib::error::ErrorDomain;
+use glib::translate::*;
+"#));
+
+        for (_, type_) in env.library.namespace_types(namespaces::MAIN) {
+            if let Type::Enumeration(ref enum_) = *type_ {
+                try!(generate_enum(env, w, enum_));
+                if let Some (cfg) = version_condition_string(env.config.min_cfg_version,
```

Comment:
Too many calls `version_condition_string` with these parameters, maybe its time to add new function with parameters `(env: &Env, version: Option<Version>)`?